crates/steel-core/src/parser/expander.rs

```diff
@@ -711,13 +731,27 @@ fn match_list_pattern(patterns: &[MacroPattern], list: &[ExprKind], improper: bo
                 }
                 continue;
             }
-            MacroPattern::ManyNested(subpatterns) if has_ellipsis => {
+            MacroPattern::ManyNested(subpatterns, is_vec) if has_ellipsis => {
                 for _ in 0..expected_many_captures {
-                    let Some((_, ExprKind::List(l))) = exprs_iter.next() else {
+                    let Some((_, next)) = exprs_iter.next() else {
                         return false;
                     };
 
-                    if !match_list_pattern(subpatterns, &l.args, l.improper) {
+                    // FIXME: this is not quite correct. It does not handle
```

Comment:
I wanted to fix this in this PR, but it's already sizeable and near overflowing my head capacity at the moment.